### PR TITLE
Start moving documentation to JAX

### PR DESCRIPTION
**Pull request by gautierronan**
First pass on the documentation. Had to remove a bunch of things that are not yet supported (e.g. smesolve or QuTiP and list objects as inputs). Still need to go over the Python API, but otherwise I think it's ok-ish for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,86 +34,73 @@ pip install git+https://github.com/dynamiqs/dynamiqs.git
...
+    tsave = jnp.linspace(0, 1.0, 101)
+
+    # run simulation
+    result = dq.mesolve(H, jump_ops, psi0, tsave)
+
+    return dq.expect(dq.dag(a) @ a, result.states[-1]).real
+
+# compute gradient with respect to kappa and alpha
+grad_population = jax.grad(population, argnums=(1, 2))
+print(grad_population(omega, kappa, alpha))
```
Comment (pierreguilmin) [inline README.md:98]: We need a nicer and clearer output, e.g.:
```suggestion
grad_population = jax.grad(population, argnums=(1, 2))
grads = grad_population(omega, kappa, alpha)
print(f'Gradient w.r.t. kappa={grads[1]:.2f}, alpha={grads[2]:.2f}.')
```

File: docs/getting_started/sharp-bits.md [hunk 2]
```diff
@@ -37,54 +35,31 @@ In QuTiP, the `*` symbol is used to multiply two operators. This convention also
...
-tensor([[0.+0.j, 1.+0.j],
-        [1.+0.j, 0.+0.j]])
+Array([[0.+0.j, 1.+0.j],
+       [1.+0.j, 0.+0.j]], dtype=complex64)
 ```
 
 ### Computing the adjoint
 
-Use `dq.dag(x)`, `x.mH` or `x.adjoint()` instead of `x.dag()` to get the hermitian conjugate of `x`.
+Use `dq.dag(x)` or `x.T.conj()` instead of `x.dag()` to get the hermitian conjugate of `x`.
```
Comment (pierreguilmin) [inline docs/getting_started/sharp-bits.md:58]:
```suggestion
Use `dq.dag(x)` or `x.mT.conj()` instead of `x.dag()` to get the hermitian conjugate of `x`.
```

File: README.md [hunk 3]
```diff
@@ -36,86 +34,73 @@ pip install git+https://github.com/dynamiqs/dynamiqs.git
...
-# QuTiP operators, initial state and saving times
-a = qt.destroy(n)
-H = omega * a.dag() * a
-jump_ops = [np.sqrt(kappa) * a]
-psi0 = qt.coherent(n, alpha0)
-tsave = np.linspace(0, 1.0, 101)
-
-# run on GPU if available, otherwise on CPU
-torch.set_default_device('cuda' if torch.cuda.is_available() else 'cpu')
+# initialize operators, state and saving times
```
Comment (pierreguilmin) [inline README.md:49]:
```suggestion
# initialize operators, initial state and saving times
```
And in the other example too

File: README.md [hunk 4]
```diff
@@ -36,86 +34,73 @@ pip install git+https://github.com/dynamiqs/dynamiqs.git
...
-loss.backward()
-print(kappa.grad)
-print(alpha0.grad)
+n = 128      # Hilbert space dimension
+omega = 1.0  # frequency
+kappa = 0.1  # decay rate
+alpha = 1.0  # initial coherent state amplitude
+
+def population(omega, kappa, alpha):
+    """Return the population inside the cavity after time evolution."""
```
Comment (pierreguilmin) [inline README.md:83]:
```suggestion
    """Return the oscillator population after time evolution."""
```

File: docs/getting_started/sharp-bits.md [hunk 5]
```diff
@@ -17,17 +14,18 @@ The syntax in dynamiqs is similar to [QuTiP](http://qutip.org/), a popular Pytho
 
 ### Adding a scalar to an operator
 
-In QuTiP, adding a scalar to a `QObj` performs an implicit multiplication of the scalar with the identity matrix. This convention differs from the one adopted by common scientific libraries such as NumPy and PyTorch. In dynamiqs, adding a scalar to a tensor performs an element-wise addition. To achieve the same result as in QuTiP, you must **explicitly multiply the scalar with the identity matrix**:
+In QuTiP, adding a scalar to a `QObj` performs an implicit multiplication of the scalar with the identity matrix. This convention differs from the one adopted by common scientific libraries such as NumPy, PyTorch, SciPy or JAX. In dynamiqs, adding a scalar to a tensor performs an element-wise addition. To achieve the same result as in QuTiP, you must **explicitly multiply the scalar with the identity matrix**:
```
Comment (pierreguilmin) [inline docs/getting_started/sharp-bits.md:17]:
```suggestion
In QuTiP, adding a scalar to a `QObj` performs an implicit multiplication of the scalar with the identity matrix. This convention differs from the one adopted by common scientific libraries such as NumPy, PyTorch, SciPy or JAX. In dynamiqs, adding a scalar to an array performs an element-wise addition. To achieve the same result as in QuTiP, you must **explicitly multiply the scalar with the identity matrix**:
```
ctrl f + tensor will higlight all similar occurences that need to be fixed

File: docs/tutorials/defining-hamiltonians.md [hunk 6]
```diff
@@ -1,36 +1,28 @@
 # Defining Hamiltonians
 
-In this short tutorial, we explain how to define Hamiltonians in dynamiqs. There are currently three ways: using array-like objects for constant Hamiltonians, defining a function for time-dependent Hamiltonians, and using a custom list format for piecewise constant Hamiltonians.
+In this short tutorial, we explain how to define Hamiltonians in dynamiqs. There are currently four ways: using array-like objects for constant Hamiltonians, defining a function or function-like coefficients for time-dependent Hamiltonians, and using a custom format for piecewise constant Hamiltonians.
```
Comment (pierreguilmin) [inline docs/tutorials/defining-hamiltonians.md:3]: "defining a function or function-like coefficients for time-dependent Hamiltonians" > unclear, I think you can split both cases with a coma and explain individually.

File: docs/getting_started/sharp-bits.md (hunk 2 above)
Comment (pierreguilmin) [reply docs/getting_started/sharp-bits.md:58]: And at other occurences too.

File: docs/tutorials/open-system.md [hunk 7]
```diff
@@ -104,19 +104,20 @@ Also called the **quantum-jump** approach.
 You can create the state, Hamiltonian and jump operators using any array-like object (Python lists, NumPy arrays, QuTiP quantum objects or PyTorch tensors). Let's take the example of a two-level system with a simple Hamiltonian and a single jump operator:
```
Comment (pierreguilmin) [inline docs/tutorials/open-system.md:104]: Remove types that don't work as you did for closed system.

File: docs/tutorials/workflow.md [hunk 8]
```diff
@@ -36,45 +35,38 @@ print(f"Hamiltonian of type {type(H)} and shape {H.shape}.")
...
+State of type <class 'jaxlib.xla_extension.ArrayImpl'> and shape (2, 1).
+Hamiltonian of type <class 'jaxlib.xla_extension.ArrayImpl'> and shape (2, 2).
 ```
 
-In dynamiqs, all quantum objects are defined directly with PyTorch [Tensors](https://pytorch.org/docs/stable/tensors.html), and without subclassing. This allows for easy interfacing with PyTorch utility functions, and avoids runtime overheads. Also, all quantum objects have at least two dimensions to avoid systematic reshaping or coding mistakes (e.g. trying to multiply a ket and an operator in the wrong order).
-In particular, kets have a shape `(..., N, 1)`.
+In dynamiqs, all quantum objects are defined directly with [JAX arrays](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.array.html), and without subclassing. This allows for easy interfacing with JAX utility functions, and avoids runtime overheads. Also, all quantum objects have at least two dimensions to avoid systematic reshaping or coding mistakes (e.g. trying to multiply a ket and an operator in the wrong order). In particular, kets have a shape `(..., N, 1)`. Note also that objects are single-precision (`float32` or `complex64`) by default.
```
Comment (pierreguilmin) [inline docs/tutorials/workflow.md:42]:
```suggestion
In dynamiqs, all quantum objects are defined directly with [JAX arrays](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.array.html), and without subclassing. This allows for easy interfacing with JAX utility functions, and avoids runtime overheads. Also, all quantum objects have at least two dimensions to avoid systematic reshaping or coding mistakes (e.g. trying to multiply a ket and an operator in the wrong order). In particular, kets have a shape `(..., n, 1)`. Note also that objects are single-precision (`float32` or `complex64`) by default.
```

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Beautiful, thanks a lot!! Nit comments only.

File: README.md [hunk 9]
```diff
@@ -6,23 +6,21 @@
...
 Some exciting features of dynamiqs include:
 
-- Running simulations on **GPUs**, with a significant speedup for large Hilbert space dimensions.
+- Running simulations on **GPUs** and **TPUs**, with a significant speedup for large Hilbert space dimensions.
 - **Batching** many simulations of different Hamiltonians, jump operators or initial states to run them concurrently.
 - Exploring solvers **tailored to quantum** simulations that preserve the properties of the state, such as trace and positivity.
 - Computing **gradients** of any function of the evolved quantum state with respect to any parameter of the Hamiltonian, jump operators, or initial state.
-- Using the library as a drop-in replacement for [QuTiP](https://qutip.org/) by directly passing QuTiP-defined quantum objects to our solvers.
-- Implementing your own solvers with ease by subclassing our base solver class and focusing directly on the solver logic.
-- Enjoy reading our carefully crafted documentation on our website: <https://www.dynamiqs.org>.
```
[the remark this answers is not in the document]
Comment (gautierronan) [reply README.md:23]: For the subclassing, it was not so clear to me whether we still could. The documentation is a mistake: I will put it back on.

[the remark this answers is not in the document]
Comment (pierreguilmin) [reply README.md:23]: Yes we can! (just implement `SESolver` run method) I plan to write a tutorial "Write my own solver" someday (e.g. to do a simple Euler), I think it's a bit selling point of the library (as compared to QuTiP for example) that you can write solvers in plain Python and test them very quickly.

File: README.md [hunk 10]
```diff
@@ -36,86 +34,73 @@ pip install git+https://github.com/dynamiqs/dynamiqs.git
...
 print(result)
 ```
 
 ```text
-|██████████| 100.0% - time 00:00/00:00
 ==== Result ====
-Method       : Dopri5
-Start        : 2023-09-10 16:57:34
-End          : 2023-09-10 16:57:35
-Total time   : 0.48 s
```
Comment (abocquet) [inline README.md:74]: Note for later, we could add these informations again

File: docs/getting_started/sharp-bits.md (hunk 5 above)
Comment (abocquet) [inline docs/getting_started/sharp-bits.md:17]:
```suggestion
In QuTiP, adding a scalar to a `QObj` performs an implicit multiplication of the scalar with the identity matrix. This convention differs from the one adopted by common scientific libraries such as NumPy, PyTorch or JAX. In dynamiqs, adding a scalar to a tensor performs an element-wise addition. To achieve the same result as in QuTiP, you must **explicitly multiply the scalar with the identity matrix**:
```
scipy uses numpy so it has the same conventions

File: docs/tutorials/batching-simulations.md [hunk 11]
```diff
@@ -123,24 +122,24 @@ So we want to run a total of `11 * 11 = 121` simulations. Let's compare how long
...
->>> %timeit run_batched('cuda')
-20.1 ms ± 78.7 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
+>>> %timeit run_unbatched()
+439 ms ± 692 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
+>>> %timeit run_batched()
+6.29 ms ± 160 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
 ```
 
-On the GPU, because we save costly data transfers with the CPU and do N-D matrices multiplications, we gain a **factor x75** in speedup!
+On the GPU, because we save costly data transfers with the CPU and do N-D matrices multiplications, we gain a **factor x70** in speedup!
```
Comment (abocquet) [inline docs/tutorials/batching-simulations.md:143]: There is not much data transfer involved that we remove, only latency. This phrase should be corrected or removed

File: README.md (hunk 10 above)
Comment (gautierronan) [reply README.md:74]: Indeed !

File: docs/getting_started/sharp-bits.md (hunk 5 above)
Comment (gautierronan) [reply docs/getting_started/sharp-bits.md:17]: I agree, but thought it would make the point stronger, no? Can remove TBH.

Comment (abocquet) [reply docs/getting_started/sharp-bits.md]: Pytorch, JAX and Numpy already make a fairly strong point

**Review by pierreguilmin — DISMISSED** [review]

**Review by pierreguilmin — DISMISSED** [review]
That's great 😍

**Review by pierreguilmin — APPROVED** [review]
